src/libs/actions/BankAccounts.js

```diff
@@ -339,9 +339,9 @@ function fetchFreePlanVerifiedBankAccount(stepToOpen) {
     // We are using set here since we will rely on data from the server (not local data) to populate the VBA flow
     // and determine which step to navigate to.
     Onyx.set(ONYXKEYS.REIMBURSEMENT_ACCOUNT, {
-        loading: true,
+        loading: true, // shows loading indicator
         error: '',
-
+        fetching: true, // only used when fetching
```

Comment:
Why do we need both `fetching` and `loading`? The comments here don't really explain the difference or when we'd want to refer to one or the other or why.

Reply:
This doesn't feel that clean to me either, but the purpose is this:

I think we were doing the second fetch just with the purpose of reacting to the route parameter and show the right screen when we open directly a route like `/bank-account/company`

Since I'm avoiding to fetch the information a second time in the `ReimbursementAccountPage`'s constructor, I needed a way to know that the first fetch had finished to trigger the check about the route parameter ([here](https://github.com/Expensify/App/blob/251ee9719c37a3bd0259c89a31914bec0c964295/src/pages/ReimbursementAccount/ReimbursementAccountPage.js#L106))

I didn't use the `loading` state because that is used to [show the loading indicator](https://github.com/Expensify/App/blob/main/src/pages/ReimbursementAccount/ReimbursementAccountPage.js#L176-L178), and it is not used only when fetching, we also set it to `loading: true` when submitting information to the server. I don't want to trigger the [code related to the route parameter](https://github.com/Expensify/App/blob/251ee9719c37a3bd0259c89a31914bec0c964295/src/pages/ReimbursementAccount/ReimbursementAccountPage.js#L106)) when we submit information.

Is my assumption correct about this code only being there to handle the application starting from one of these routes? If I remove that code, everything seems to work fine, except for if I open the application from `/bank-account/company`, I get the screen to choose between manual and plaid.

Reply:
Removed this, if we decide to work in removing the second fetch, it will be in another PR.